File: 1447.最简分数.py
#
# @lc app=leetcode.cn id=1447 lang=python3
#
# [1447] 最简分数
#

# @lc code=start
class Solution:
    def simplifiedFractions(self, n: int) -> List[str]:
        res = []
        for down in range(2, n + 1):
            for up in range(down):
                if self.gcd(up, down) == 1:
                    res.append(f"{up}/{down}")
        return res
    
    def gcd(self, x, y):
        while x != 0:
            x, y = y % x, x
        return y
# @lc code=end

File: 1447.最简分数.go
/*
 * @lc app=leetcode.cn id=1447 lang=golang
 *
 * [1447] 最简分数
 */

// @lc code=start
func simplifiedFractions(n int) (res []string) {
	for down := 2; down < n + 1; down++ {
		for up := 1; up < down; up++ {
			if gcd(up, down) == 1 {
				res = append(res, strconv.Itoa(up) + "/" + strconv.Itoa(down))
			}
		}
	}
	return
}


func gcd(x, y int) int {
	for x != 0 {
		x, y = y % x, x
	}
	return y
}
// @lc code=end

